### PR TITLE
Test leaderboard get static info

### LINKED ISSUE #612: test: `/leaderboard/get_static_info?addr=${addr}&duration=${duration}`
read description here - https://github.com/starknet-id/starknet.quest/issues/602#issue-2255907921

### PR DESCRIPTION
**Pull request by Ugo-X**
Resolves: #612

Pull Request type

Testing
Resolves: #612

### CODE REVIEW

File: tests/services/leaderboard.test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { fetchLeaderboardToppers } from "@services/apiService";
```
Comment (ayushtom) [inline tests/services/leaderboard.test.js]: please include in the apiService.test.js

File: tests/services/leaderboard.test.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+const API_URL = process.env.NEXT_PUBLIC_API_LINK;
+global.fetch = jest.fn();
+
+describe("fetchLeaderboardToppers", () => {
+  afterEach(() => {
+    fetch.mockClear();
+  });
+
+  it("fetches leaderboard toppers correctly", async () => {
+    const mockResponse = { leaderboardData: "mockedData" };
```
Comment (ayushtom) [inline tests/services/leaderboard.test.js]: i dont think this is the correct response.Can you please check with the api and add that data here?

File: tests/services/leaderboard.test.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    const params = { addr: "exampleAddr", duration: "exampleDuration" };
+
+    fetch.mockResolvedValueOnce({
+      json: () => Promise.resolve(mockResponse),
+    });
+
+    const response = await fetchLeaderboardToppers(params);
+    expect(fetch).toHaveBeenCalledWith(
```
Comment (ayushtom) [inline tests/services/leaderboard.test.js]: this call should actually return - Invalid duration

File: tests/services/leaderboard.test.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      `${API_URL}/leaderboard/get_static_info?addr=${params.addr}&duration=${params.duration}`
+    );
+    expect(response).toEqual(mockResponse);
+  });
+
+  it("handles errors gracefully", async () => {
```
Comment (ayushtom) [inline tests/services/leaderboard.test.js]: you also need to add sending null and undefined values here

**Comment by ayushtom** [comment]
Please resolve conflicts too and add the response format to the api call assigned to you so that we have those typings on the front

**Comment by Ugo-X** [comment]
I have it done @ayushtom 
Screenshot 2024-04-26 at 18 15 08

**Review by ayushtom — APPROVED** [review]
lgtm

**Pull request merged**